src/Microsoft.Extensions.CodeGeneration.Core/DefaultCodeGeneratorAssemblyProvider.cs

```diff
@@ -54,8 +51,12 @@ public IEnumerable<Assembly> CandidateAssemblies
                     .SelectMany(_libraryManager.GetReferencingLibraries)
                     .Distinct()
                     .Where(IsCandidateLibrary);
-                
+#if NET451
+                // For Net 451 the exe is being invoked from the build folder so all dependencies are next to the exe.
+                return list.Select(lib => _assemblyLoadContext.LoadFromName(new AssemblyName(lib.Identity.Name)));
```

Comment:
The real reason you don't get an error is that the tool is pulling in Newtonsoft.Json 8. https://github.com/aspnet/Scaffolding/blob/81d4b51c6282df5ea2d185a2936dcfb4dba284d9/src/Microsoft.Extensions.CodeGeneration.Core/project.json#L12 Use caution with `AssemblyLoadContext`. We are ditching it completely for dotnet-ef because it doesn't provide the isolation we actually need to execute user code.